incubator/hnc/pkg/kubectl/configset.go

```diff
@@ -0,0 +1,65 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	api "github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/api/v1alpha1"
+)
+
+var setTypeCmd = &cobra.Command{
+	Use: fmt.Sprintf("set-type <%s|%s|%s>", string(api.Propagate), string(api.Remove),
+		string(api.Ignore)),
+	Short: "Sets the HNC configuration of a specific type",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		mode := api.SynchronizationMode(args[0])
+		flags := cmd.Flags()
+		apiVersion, _ := flags.GetString("apiVersion")
+		kind, _ := flags.GetString("kind")
+		config := client.getHNCConfig()
+
+		exist := false
+		for i := 0; i < len(config.Spec.Types); i++ {
+			t := &config.Spec.Types[i]
+			if t.APIVersion == apiVersion && t.Kind == kind {
+				t.Mode = mode
+				exist = true
+				break
+			}
+		}
+
+		if !exist {
+			config.Spec.Types = append(config.Spec.Types,
+				api.TypeSynchronizationSpec{
+					APIVersion: apiVersion,
+					Kind:       kind,
+					Mode:       mode,
+				})
+		}
+
+		client.updateHNCConfig(config)
+	},
+}
+
+func newSetTypeCmd() *cobra.Command {
+	setTypeCmd.Flags().String("apiVersion", "", "API version of the kind")
```

Comment:
Some examples here might be nice, e.g. "v1" for core objects or "example.com/v1beta1" for a contributed type.

Reply:
Thanks for the suggestion! Done. PTAL.